frontend/src/utils/apiService/AuthProvider.js

```diff
@@ -0,0 +1,25 @@
+import { createContext, useState } from 'react'
+import axios from 'axios'
+import { useHistory } from 'react-router-dom'
+
+export const AuthContext = createContext({})
+
+function AuthProvider({ children }) {
+    const [token, setToken] = useState()
+    const history = useHistory()
+
+    const login = credentials => {
+        axios
+            .post('/auth/login', credentials)
+            .then(response => response.data)
+            .then(setToken)
+            .then(() => history.push('/'))
+            .catch(error => console.error(error.message))
```

Comment:
you might wanna do something her thet lets the user know that a wrong password had been typed